src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -507,13 +506,7 @@ private void initializeConnections()
    {
       if (config.isInitializationFailFast()) {
          try {
-            Connection connection = getConnection();
-            if (config.getMinimumIdle() == 0) {
-               evictConnection(connection);
-            }
-            else {
-               connection.close();
-            }
+            newConnection().close();
```

Comment:
This is less efficient.  This will create a connection, and then close it.  The existing code will only close the connection if ``minimumIdle=0``, otherwise it will be returned to the pool.